sys/posix/sockets/posix_sockets.c

```diff
@@ -170,110 +206,73 @@ static inline int _get_data_from_sockaddr(const struct sockaddr *address, size_t
                 return -1;
             }
             struct sockaddr_in *in_addr = (struct sockaddr_in *)address;
-            *addr = &in_addr->sin_addr;
-            *addr_len = sizeof(ipv4_addr_t);
-            port->u16 = in_addr->sin_port;
+            out->family = AF_INET;
+            out->addr.ipv4 = in_addr->sin_addr.s_addr;
+            out->port = ntohs(in_addr->sin_port);
             break;
+#ifdef SOCK_HAS_IPV6
```

Comment:
Well in sock it is that way, because it is not just build for networks we target but also as a simplification for POSIX sockets (see [kaspar030/sock](https://github.com/kaspar030/sock/tree/master/src/posix)). And from an application point of view it does make sense, since IPv4 is always integrated in some capacity in IPv6 while the reverse is not the case + IPv6 needs more memory (due to its long addresses) so optimizing it out if necessary is good.